packages/flutter/lib/src/widgets/sliver.dart

```diff
@@ -388,6 +410,29 @@ class SliverChildBuilderDelegate extends SliverChildDelegate {
   /// Defaults to providing an index for each widget.
   final SemanticIndexCallback semanticIndexCallback;
 
+  /// Called to find the new index of a child based on its key in case of reordering.
+  ///
+  /// If not provided, a child widget may not map to its existing [RenderObject]
+  /// when the order in which children are returned from [builder] changes.
+  /// This may result in state-loss.
+  final Function findChildIndex;
+
+  @override
+  int findIndexByKey(Key key) {
+    if (findChildIndex == null)
+      return -1;
+    assert(key != null);
+    Key childKey;
+    if (key is _SaltedValueKey) {
+      // in this case, child key is wrapped with _SaltedValueKey
```

Comment:
nit: in -> In and `.` at the end.

Reply:
Actually, the comment doesn't seem to provide too much value. I would probably just remove it.